plugins/woocommerce/client/blocks/assets/js/blocks/product-description/edit.tsx

```diff
@@ -0,0 +1,200 @@
+		 *
+		 * The preview using the raw blocks can only be presented to users with
+		 * edit permissions for the post to prevent potential exposure of private
+		 * block content.
+		 */
+		return <div { ...blockPreviewProps }></div>;
+	}
+
+	return content?.protected ? (
+		<div { ...blockProps }>
+			<Warning>
+				{ __( 'This content is password protected.', 'woocommerce' ) }
+			</Warning>
+		</div>
+	) : (
+		<div
+			{ ...blockProps }
+			dangerouslySetInnerHTML={ { __html: content?.rendered } }
+		></div>
+	);
+}
+
+function EditableContent( { context = {} } ) {
+	const { postType, postId } = context;
+
+	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
+		'postType',
+		postType,
+		{ id: postId }
+	);
+
+	const entityRecord = useSelect(
+		( select ) => {
+			return select( coreStore ).getEntityRecord(
+				'postType',
+				postType,
+				postId
+			);
+		},
+		[ postType, postId ]
+	);
+
+	const hasInnerBlocks = !! entityRecord?.content?.raw || blocks?.length;
+
+	const initialInnerBlocks = [ [ 'core/paragraph' ] ];
+
+	const props = useInnerBlocksProps(
+		useBlockProps( { className: 'product-description__editable-content' } ),
+		{
+			value: blocks,
+			onInput,
+			onChange,
+			template: ! hasInnerBlocks ? initialInnerBlocks : undefined,
+		}
+	);
+	return <div { ...props } />;
+}
+
+function Content( props ) {
+	const { context: { postType, postId } = {}, layoutClassNames } = props;
+
+	const userCanEdit = useCanEditEntity( 'postType', postType, postId );
+	if ( userCanEdit === undefined ) {
+		return null;
+	}
+
+	return userCanEdit ? (
+		<EditableContent { ...props } />
+	) : (
+		<ReadOnlyContent
+			parentLayout={ props.parentLayout }
+			layoutClassNames={ layoutClassNames }
+			userCanEdit={ userCanEdit }
+			postType={ postType }
+			postId={ postId }
+		/>
+	);
+}
+
+function Placeholder( { layoutClassNames } ) {
+	const blockProps = useBlockProps( { className: layoutClassNames } );
+	return (
+		<div { ...blockProps }>
+			<p>
+				{ __(
+					'This block displays the product description. When viewing a product page, the description content will automatically appear here.',
+					'woocommerce'
+				) }
+			</p>
+		</div>
+	);
+}
+
+function RecursionError() {
+	const blockProps = useBlockProps();
+	return (
+		<div { ...blockProps }>
+			<Warning>
+				{ __(
+					'Block cannot be rendered inside itself.',
+					'woocommerce'
+				) }
+			</Warning>
+		</div>
+	);
+}
+
+export default function ProductDescriptionEdit( {
+	context,
+	__unstableLayoutClassNames: layoutClassNames,
+	__unstableParentLayout: parentLayout,
+} ) {
+	const { postId: contextPostId, postType: contextPostType } = context;
+	const hasAlreadyRendered = useHasRecursion( contextPostId );
+
+	if ( contextPostId && contextPostType && hasAlreadyRendered ) {
+		return <RecursionError />;
+	}
+
+	return (
+		<RecursionProvider uniqueId={ contextPostId }>
+			{ contextPostId && contextPostType ? (
```

Comment:
I think we should update this condition to be product specific? and/or update the product details block to be limited by the single-product, product-template, and post-template parent ancestors.
Mostly because I can cause this funny bug where if I add the product details ( with the product description auto added ) to the top level of a page or post it will render the content of the page and post, causing this weird recursion:
<img width="320" alt="Screenshot 2025-03-12 at 4 27 11 PM" src="https://github.com/user-attachments/assets/4bb15da4-4389-4a2d-b406-47642a92ce9f" />

Reply:
Maybe the most reliable way is to make use of the product data store and use `select( productsStore ).getProduct( contextPostId )` and see if it returns a product? instead of checking the post type specifically?

Reply:
> Mostly because I can cause this funny bug where if I add the product details ( with the product description auto added ) to the top level of a page or post it will render the content of the page and post, causing this weird recursion:

IMO Similar to Product Description block, we should restrict Product Details block to be available only in Product Collection block, Single Product block, or other product-aware blocks. In my testing, If I add Product Details block then Product Description block doesn't work at all, because it's in invalid context. Isn't that happening for you as well?

IMO It should be safe to assume that if Product Description block is inside Product Details block then it will have the product context. What do you think?